oak-core/src/main/java/org/apache/jackrabbit/oak/Oak.java

```diff
@@ -576,6 +577,8 @@ public Oak with(@NotNull Whiteboard whiteboard) {
             newSettings.setLimitReads(this.queryEngineSettings.getLimitReads());
             this.queryEngineSettings = new AnnotatedQueryEngineSettings(newSettings);
         }
+        Feature prefetchFeature = Feature.newFeature(QueryEngineSettings.FT_NAME_PREFETCH_FOR_QUERIES, whiteboard);
+        queryEngineSettings.setPrefetchFeature(prefetchFeature);
```

Comment:
This seems to be causing an NPE in one of the tests in oak-core.

Reply:
Why do you set this feature here? Wouldn't it be sufficient to centralize this handling in the ```QueryEngine``` itself and just there?

Reply:
> Why do you set this feature here?
Because "whiteboard" is not available in the QueryEngine.

Reply:
@nit0906 NPE: you are right! It is now resolved.